docs/configuration/secure.md

```diff
@@ -0,0 +1,614 @@
+# Table of Contents
+- [Hardening and Security Guidance](#hardening-and-security-guidance)
+  * [List of placeholder names](#list-of-placeholder-names)
+  * [Firewalling](#firewalling)
+  * [File Access Rights](#file-access-rights)
+  * [Place data directory outside of the web root](#place-data-directory-outside-of-the-web-root)
+  * [Isolate Docroot](#isolate-docroot)
+  * [OS user handling security](#os-user-handling-security)
+  * [Major security improvement: use HTTPS](#major-security-improvement--use-https)
+    + [Redirect all unencrypted traffic to HTTPS](#redirect-all-unencrypted-traffic-to-https)
+    + [Proper SSL configuration](#proper-ssl-configuration)
+    + [Serve security related Headers](#serve-security-related-headers)
+    + [Enable OCSP stapling (TLS Certificate Status Request)](#enable-ocsp-stapling--tls-certificate-status-request-)
+    + [Enable HTTP Strict Transport Security](#enable-http-strict-transport-security)
+    + [Genrate the ILIAS ```ilClientId``` cookie with ```secure``` attribute](#genrate-the-ilias----ilclientid----cookie-with----secure----attribute)
+  * [Suppress server signature and PHP version information](#suppress-server-signature-and-php-version-information)
+  * [deny access or restrict to several files or locations](#deny-access-or-restrict-to-several-files-or-locations)
+    + [ILIAS setup](#ilias-setup)
+    + [Prevent blacklisted files of beeing served by the webserver](#prevent-blacklisted-files-of-beeing-served-by-the-webserver)
+    + [Prevent execution of PHP-Code in data-directory](#prevent-execution-of-php-code-in-data-directory)
+    + [Deny Access to local Git-Directory](#deny-access-to-local-git-directory)
+  * [Integrity check of ILIAS code in docroot](#integrity-check-of-ilias-code-in-docroot)
+  * [Use WebAccessChecker](#use-webaccesschecker)
+  * [Use secure passwords](#use-secure-passwords)
+  * [Report security issues](#report-security-issues)
+
+# Hardening and Security Guidance
+
+This guideline is going to show you some best practice examples how to improve the security of your ILIAS installation, of the webserver and it's associated components.
+The ILIAS e.V. requested a documentation for a "Secure ILIAS" in march 2019.
+The section "Hardening and Security Guidance" should be removed and the security related instructions have to be mantained in a own document.
+
+## List of placeholder names
+
+In the following text includes some placeholder.
+For a better identification, they will describe here:
+
+|  placeholder |  description |
+|-------|-------|
+|  %USERNAME% |  a choiceable user name or the webserver user based on distribution  |
```

Comment:
The correct word would be "choosable", although I would suggest to use individual
```suggestion
| %USERNAME% |  an individual user name or the webserver user based on distribution  |
```